CalibCalorimetry/HcalAlgos/src/HcalPulseShapes.cc

```diff
@@ -806,15 +831,58 @@ double HcalPulseShapes::analyticPulseShapeSiPMHE(double t) {
     onePulse(t,A2,sigma2_shape,theta2_loc,m2_scale);
 }
 
-double HcalPulseShapes::generatePhotonTime(CLHEP::HepRandomEngine* engine) {
+double HcalPulseShapes::generatePhotonTime(CLHEP::HepRandomEngine* engine, unsigned int signalShape) {
+  if(signalShape==206) return generatePhotonTime206(engine);
+  else return generatePhotonTime203(engine);
+}
+
+double HcalPulseShapes::generatePhotonTime203(CLHEP::HepRandomEngine* engine) {
   double result(0.);
   while (true) {
     result = CLHEP::RandFlat::shoot(engine, HcalPulseShapes::Y11RANGE_);
-    if (CLHEP::RandFlat::shoot(engine, HcalPulseShapes::Y11MAX_) < HcalPulseShapes::Y11TimePDF(result))
+    if (CLHEP::RandFlat::shoot(engine, HcalPulseShapes::Y11MAX203_) < HcalPulseShapes::Y11203(result))
       return result;
   }
 }
 
-double HcalPulseShapes::Y11TimePDF(double t) {
+double HcalPulseShapes::generatePhotonTime206(CLHEP::HepRandomEngine* engine) {
+  double result(0.);
+  while (true) {
+    result = CLHEP::RandFlat::shoot(engine, HcalPulseShapes::Y11RANGE_);
+    if (CLHEP::RandFlat::shoot(engine, HcalPulseShapes::Y11MAX206_) < HcalPulseShapes::Y11206(result))
+      return result;
+  }
+}
+
+//Original scintillator+Y11 fit from Vasken's 2001 measurement
+double HcalPulseShapes::Y11203(double t) {
   return exp(-0.0635-0.1518*t)*pow(t, 2.528)/2485.9;
 }
+
+//New scintillator+Y11 model from Vasken's 2017 measurement plus a Landau correction term
```

Comment:
could you add a link to this documentation?

Reply:
If modify this place I would use
return exp(-0.0635-0.1518*t + log(t)*2.528)/2485.9;
which seems to be faster

Reply:
https://indico.cern.ch/event/678759/contributions/2788320/attachments/1570351/2476966/DPG-Reco-CMSWeek-Dec2017.pdf
Here is a link to the talk with information about the measurement? Do you want me to add it as a comment in the PR?